request.go

```diff
@@ -118,21 +118,28 @@ func checkAndConvertOptions(root *Command, opts OptMap, path []string) (OptMap,
 
 		kind := reflect.TypeOf(v).Kind()
 		if kind != opt.Type() {
-			if str, ok := v.(string); ok {
-				val, err := opt.Parse(str)
-				if err != nil {
-					value := fmt.Sprintf("value %q", v)
-					if len(str) == 0 {
-						value = "empty value"
-					}
-					return options, fmt.Errorf("could not convert %s to type %q (for option %q)",
-						value, opt.Type().String(), "-"+k)
+			if opt.Type() == Strings {
+				if _, ok := v.([]string); !ok {
+					return options, fmt.Errorf("option %q should be type %q, but got type %q",
+						k, opt.Type().String(), kind.String())
 				}
-				options[k] = val
-
 			} else {
-				return options, fmt.Errorf("option %q should be type %q, but got type %q",
-					k, opt.Type().String(), kind.String())
+				if str, ok := v.(string); ok {
```

Comment:
We could invert this check to drop the else clause (and use an early return).

Reply:
done (although wasn't sure what to call this commit so just used "refactor: cleanup code"